controller/userController.js

```diff
@@ -86,7 +92,180 @@ const token = (req, res) => {
     }
 }
 
+const create = (req, res) => {
+    const user = new User(req.body);
+
+    // Check if an user with the provided email exists
+    User.find({ email: user.email }).then(users => {
+        if (users.length == 0) {
+            // Generate salt & passwordHash
+            let retVal = getPasswordHash(user.passwordHash);
+            user.salt = retVal.salt;
+            user.passwordHash = retVal.passwordHash;
+            // Add new user
+            user.save().then(() => {
+                user.passwordHash = null;
+                user.salt = null;
+                return res.status(HttpStatus.CREATED).json(user);
+            });
+        } else {
+            return res.status(HttpStatus.BAD_REQUEST).json({
+                status: 'Error',
+                message: 'User with provided email already exists!'
+            });
+        }
+    }).catch(err => {
+        const errorMessage = ValidationUtil.buildErrorMessage(err, 'create', 'user');
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            status: 'Error',
+            message: errorMessage
```

Comment:
You can assing value from buildErrorMessage directly since it is not used elsewhere.

Reply:
Having variables extracted at the top of the block makes the code more readable, it's just preference.